lib/archethic/replication/transaction_pool.ex

```diff
@@ -27,15 +30,31 @@ defmodule Archethic.Replication.TransactionPool do
     GenServer.cast(name, {:add_transaction, tx, validation_inputs})
   end
 
+  @doc """
+  Add the proof of validation to the queued transaction
+  """
+  @spec add_proof_of_validation(
+          gen_server :: GenServer.server(),
+          proof_of_validation :: ProofOfValidation.t(),
+          tx_address :: Crypto.prepended_hash()
+        ) :: :ok
+  def add_proof_of_validation(name \\ __MODULE__, proof_of_validation, tx_address) do
+    GenServer.cast(name, {:add_proof_of_validation, proof_of_validation, tx_address})
+  end
+
   @doc """
   Dequeue the transaction and the inputs for the given transaction's address
   """
-  @spec pop_transaction(GenServer.server(), binary()) ::
+  @spec get_transaction(
+          GenServer.server(),
+          address :: Crypto.prepended_hash(),
+          opts :: Keyword.t() | [get_transaction?: boolean()]
```

Comment:
I think you mean `[pop_transaction?: boolean]`.
Suggestion: Probably this would be clearer to keep the functions separated, like: `Map.get` `Map.pop` to avoid options